software/bwa/mem/main.nf

```diff
@@ -32,13 +31,15 @@ process BWA_MEM {
     def prefix     = options.suffix ? "${meta.id}${options.suffix}" : "${meta.id}"
     def read_group = meta.read_group ? "-R ${meta.read_group}" : ""
     """
+    INDEX=`find -L ./ -name "*.amb" | sed 's/.amb//'`
+
     bwa mem \\
         $options.args \\
         $read_group \\
         -t $task.cpus \\
-        $fasta \\
+        \$INDEX \\
         $reads \\
-        | samtools view $options.args2 -@ $task.cpus -bS -o ${prefix}.bam -
+        | samtools view $options.args2 -@ $task.cpus -bhS -o ${prefix}.bam -
```

Comment:
I think it should be
`samtools $options.args2 --threads $task.cpus -o ${prefix}.bam -`

the `view -bhS` should be part of $options.args2.

Reply:
```
  -b       output BAM
  -h       include header in SAM output
  -S       ignored (input format is auto-detected)
      --input-fmt-option OPT[=VAL]
               Specify a single input file format option in the form
               of OPTION or OPTION=VALUE
```

Those options are related to the input/output files so should be in the module. What this doesn't cater for are things like CRAM format but it's going to be another level of complexity adding that in here. Maybe save that for another day 😅